src/invariant/CacheIsConsistentWithDatabase.cpp

```diff
@@ -16,11 +17,13 @@ std::shared_ptr<Invariant>
 CacheIsConsistentWithDatabase::registerInvariant(Application& app)
 {
     return app.getInvariantManager()
-        .registerInvariant<CacheIsConsistentWithDatabase>(app.getDatabase());
+        .registerInvariant<CacheIsConsistentWithDatabase>(
```

Comment:
To think about: is it better to pass just the objects that we need or just give up and pass whole Application everywhere (as it seems that over time more and more objects are required in multiple places)?